src/components/Contacts/ContactListTableMobile.jsx

```diff
@@ -0,0 +1,213 @@
+// React Imports
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
+import IconButton from '@mui/material/IconButton';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import SendIcon from '@mui/icons-material/Send';
+import Typography from '@mui/material/Typography';
+import { useTheme } from '@mui/material/styles';
+// Custom Hooks Imports
+import useNotification from '@hooks/useNotification';
+// Util Imports
+import { saveToClipboard } from '@utils';
+// Component Imports
+import ContactProfileIcon from './ContactProfileIcon';
+
+/**
+ * @typedef {object} Contact
+ * @property {string} webId - The Web ID of the contact
+ * @property {string} givenName - The given name of the contact
+ * @property {string} familyName - The family name of the contact
+ */
+
+/**
+ * ContactListTableMobile - Component for displaying contacts in a DataGrid
+ *
+ * @memberof Contacts
+ * @name ContactListTableMobile
+ * @param {object} Props - The props for ContactListTableMobile
+ * @param {Contact[]} Props.contacts - The list of contacts to display
+ * @param {Function} Props.deleteContact - Function to delete a contact
+ * @param {Function} Props.handleSendMessage - Function to handle sending a message
+ * @returns {React.JSX.Element} The ContactListTableMobile component
+ */
+const ContactListTableMobile = ({ contacts, deleteContact, handleSendMessage }) => {
+  const { addNotification } = useNotification();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [openMenu, setOpenMenu] = useState(null);
+  const navigate = useNavigate();
+  const theme = useTheme();
+
+  const handleClick = (event, contact) => {
+    setAnchorEl(event.currentTarget);
+    setOpenMenu(contact.webId);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+    setOpenMenu(null);
+  };
+
+  const handleMenuItemClick = (action, contact) => () => {
+    action(contact);
+    handleClose();
+  };
+
+  const handleProfileClick = (contact) => {
+    // TODO: Abstract this so ContactListTableDesktop uses the same function?
```

Comment:
While `navigate` helps re-route users to the right path, it's missing two things. The `contact` state and a handler function that helps processes `contact.value` (check out the component `ContactProfileIcon`).

![Screenshot 2024-07-01 at 4 36 49 PM](https://github.com/codeforpdx/PASS/assets/14917816/f571e61b-a9b6-4348-92ca-c4680ad77929)

Reply:
What's the optimal way of implementing this? I'm trying not to mess with ContactProfileIcon if at all possible.

Reply:
Adding the contact data to the state option of the `navigate` function appears to fix the issue of routing to the wrong contact:
```
navigate(`/contacts/${encodeURIComponent(contact.webId)}`, {state: {contact}});
```

Reply:
Yep, it's as @dbmzzo stated, there's an option to pass in the state in `navigate` with React Router (https://reactrouter.com/en/main/hooks/use-navigate#optionsstate).

In our case, we would pass in `{ state: { contact: contact.value } }` after the path. As for the `onClick`, you would want to use the same handler function that ContactProfileIcon uses and have that run before `navigate`

Reply:
I was trying testing out `navigate(`/contacts/${encodeURIComponent(contact.webId)}`, {state: {contact}});` earlier but can for sure keep trying.

Reply:
Yeah try with `{ state: { contact: contact.value } }` instead. It might be looking for a state that's just part of contact and not the full contact object itself.